pkg/utils/tekton/controller.go

```diff
@@ -364,17 +364,34 @@ func findCosignResultsForImage(imageRef string, client crclient.Client) (*Cosign
 	imageNameInfo := strings.Split(imageInfo[2], "@")
 	imageStreamName, imageDigest := imageNameInfo[0], imageNameInfo[1]
 
-	tags := &unstructured.UnstructuredList{}
-	tags.SetGroupVersionKind(schema.GroupVersionKind{
+	istGVK := schema.GroupVersionKind{
 		Group:   "image.openshift.io",
 		Kind:    "ImageStreamTag",
 		Version: "v1",
-	})
+	}
+
+	tags := &unstructured.UnstructuredList{}
```

Comment:
I realize now that this is a very inefficient way of doing things. Instead of fetching a list of all the ImageStreamTags, then fetching details for each one of those, we should just change `findTagWithName` and `findTagWithName` to query the cluster to find the exact resource we're looking for...

This is non-blocking for this PR as it does not introduce this mechanism.

Reply:
Yeah, though it's a good chance to fix this now...